### ISSUE TITLE
Determine the model architectures to explore as part of the experiments

### ISSUE THREAD

**Issue by vc1492a** [vc1492a/tidd#29]
Also important in the predictive performance is the architecture of the model. The architecture is dependent on the type of input data used, and thus this should only be explored once a better understanding of the desired input data is available (tracked in [this issue](https://github.com/vc1492a/sTEC-d-dt-Anomaly-Detection/issues/28)). 

A good base-line model may be a shallow Long Short-Term Memory Network (LSTM). 

| Index | Model Type | Layers | Optimizer | Parameter Settings | Notes | 
| ---- | ---- | ---- | ---- | ---- | ---- |
| 1 | [Long Short-Term Memory ](https://deeplearning.cs.cmu.edu/document/readings/LSTM.pdf)(LSTM) | 2 (structure TBD, but to include regularization such as dropout layers) | [Stochastic Gradient Descent](https://en.wikipedia.org/wiki/Stochastic_gradient_descent) (SGD) with decaying learning rate | TBD |  Baseline model. The decay can be performed linearly, exponentially, or by some fixed factor (2-10) each time the validation error plateaus. Includes early stopping. No batch normalization. | 
| 2 | [Long Short-Term Memory ](https://deeplearning.cs.cmu.edu/document/readings/LSTM.pdf)(LSTM) | 2 (structure TBD, but to include regularization such as dropout layers) | [Stochastic Gradient Descent](https://en.wikipedia.org/wiki/Stochastic_gradient_descent) (SGD) with decaying learning rate | TBD |  Baseline model, but use batch normalization. The decay can be performed linearly, exponentially, or by some fixed factor (2-10) each time the validation error plateaus. Includes early stopping. | 
| 3 | [Long Short-Term Memory ](https://deeplearning.cs.cmu.edu/document/readings/LSTM.pdf)(LSTM) | 2 (structure TBD, but to include regularization such as dropout layers) | [Adam](https://arxiv.org/abs/1412.6980) | TBD |  Baseline model, but with a different optimizer. Includes early stopping. No batch normalization. | 
| 4 | [Gated Recurrent Unit](https://arxiv.org/pdf/1406.1078.pdf) (GRU) | TBD | TBD | TBD | Like an LSTM, but simpler (no output gate) | 
| 5 | [Artificial Neural Network ](https://en.wikipedia.org/wiki/Artificial_neural_network#:~:text=An%20artificial%20neural%20network%20is,to%20the%20input%20of%20another.) (ANN) | TBD | TBD | TBD |  [FastAI Tabular Learner](https://dev.fast.ai/tabular.learner)  | 

**Note**: at a later stage in the project, explore time-series embeddings.
